src/dotnet/commands/dotnet-crossgen/Operations/CrossGenCmdUtil.cs

```diff
@@ -0,0 +1,180 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Tools.CrossGen.Exceptions;
+
+/// <summary>
+/// This is a wrapper of crossgen.exe native image generation utiltites
+/// </summary>
+namespace Microsoft.DotNet.Tools.CrossGen.Operations
+{
+    public class CrossGenCmdUtil
+    {
+        private enum OutputType
+        {
+            dll, pdb
+        }
+
+        private static readonly string[] s_excludedLibraries =
+        {
+            "mscorlib.dll",
+            "mscorlib.ni.dll",
+            "System.Private.CoreLib.dll",
+            "System.Private.CoreLib.ni.dll"
+        };
+
+        private readonly string _jitPath;
+        private readonly string _crossgenPath;
+        private readonly string _diaSymReaderPath;
+        private readonly NativeImageType? _outputType;
+
+        public CrossGenCmdUtil(
+            string crossgenPath,
+            string jitPath,
+            string diaSymReaderPath,
+            NativeImageType? outputType = null)
+        {
+            _crossgenPath = crossgenPath;
+            _jitPath = jitPath;
+            _diaSymReaderPath = diaSymReaderPath;
+            _outputType = outputType;
+        }
+
+        public static bool ShouldExclude(string assemblyLocation)
+        {
+            var fileName = Path.GetFileName(assemblyLocation);
+            var exclude = s_excludedLibraries.Any(lib => string.Equals(lib, fileName, StringComparison.OrdinalIgnoreCase));
+            if (exclude)
+            {
+                Reporter.Verbose.WriteLine($"Excluding {assemblyLocation} for crossgen.");
+            }
+            return exclude;
+        }
+
+        public ICollection<string> CrossGenAssembly(string appPath, string assemblyLocation, IList<string> platformAssembliesPaths, string outputDirectory, bool generateSymbols)
+        {
+            Reporter.Verbose.WriteLine($"CrossGen'ing {assemblyLocation}");
+            var outputFiles = new List<string>();
+
+            var fileName = Path.GetFileName(assemblyLocation);
+            var module = fileName.Substring(0, fileName.Length - 4);
+            var crossgenArgs = string.Join(" ", GetArgs(appPath, assemblyLocation, module, platformAssembliesPaths, outputDirectory, false));
+            var cmd = Command.Create(new CommandSpec(_crossgenPath, crossgenArgs, CommandResolutionStrategy.None))
+                .WorkingDirectory(appPath)
+                // disable partial ngen
+                .EnvironmentVariable("COMPlus_PartialNGen", "0")
+                .CaptureStdOut()
```

Comment:
It might be a bit verbose. I choose to only write it out if verbose is on if that's ok.